airflow/providers/amazon/aws/hooks/athena.py

```diff
@@ -121,7 +122,26 @@ def run_query(
         self.log.info("Query execution id: %s", query_execution_id)
         return query_execution_id
 
-    def check_query_status(self, query_execution_id: str) -> str | None:
+    def get_query_execution(self, query_execution_id: str, use_cached_response: bool = False) -> dict:
```

Comment:
Not sure if having boolean params that change API behavior is the nicest pattern but okay, I don't have a clear idea how to make it better.

Reply:
I've better not create method with same name as AthenaClient has. It might confuse end-users as S3Hook already does.
If we could select another name, it would be nice

Reply:
That's a nice suggestion. `get_query_results/info` would match?

Reply:
or `retrieve_query_execution_details`?

Reply:
Renamed to `get_query_info`.